### PR TITLE
Test reinterpreting without a range

### PR DESCRIPTION
**Pull request by ProGTX**
* Test reinterpreting without specifying range
    * To 1D
    * To same dimension with type of same size
* Minor fixes

Goes along with https://github.com/KhronosGroup/SYCL-Docs/pull/67

### CODE REVIEW

File: tests/buffer/buffer_api.cpp [hunk 1]
```diff
@@ -317,6 +424,34 @@ class TEST_NAME : public util::test_base {
...
+        sizeof(T) * inputElemsPerDim * numDims);
+    using ReinterpretT = flip_signedness_t<T>;
+
+    // Check reinterpreting with a range
+    static constexpr auto numElems = inputElemsPerDim * numDims;
+    test_buffer_reinterpret<uint8_t, T>{sizeof(T) * numElems, numElems}
+        .template check<numDims>(reinterpretInputData.data(), log);
+
+    // Check reinterpreting without a range to 1D
+    test_buffer_reinterpret_no_range<uint8_t, T, numDims, 1>::check(
```
Comment (bader) [inline tests/buffer/buffer_api.cpp:446]: To keep these tests working for SYCL-1.2.1 implementations we can surround SYCL-2020 API with
SYCL_LANGUAGE_VERSION check.

**Review by bader — CHANGES_REQUESTED** [review]
Please, ifdef SYCL-2020 API checks.

File: tests/buffer/buffer_api.cpp (hunk 1 above)
Comment (keryell) [reply tests/buffer/buffer_api.cpp:446]: Is there SYCL 2020 code in this CTS?

Comment (ProGTX) [reply tests/buffer/buffer_api.cpp:451]: This feature made it into 2020 provisional, so I've added a check for `SYCL_LANGUAGE_VERSION >= 202001`

**Review by keryell — APPROVED** [review]

**Review by bader — APPROVED** [review]
Thanks!

**Comment by bader** [comment]
@ProGTX, I think we can retarget and merge this PR to `SYCL-2020` branch. Right?

**Comment by ProGTX** [comment]
I've retargeted the branch and removed the version checks

**Review by bader — APPROVED** [review]
LGTM. One minor comment related to the updated spec.

**Comment by bader** [comment]
@keryell, does updated patch look good to you?

File: tests/buffer/buffer_api.cpp [hunk 2]
```diff
@@ -317,6 +424,34 @@ class TEST_NAME : public util::test_base {
...
+  /**
+   * @brief Tests reinterpreting a buffer
+   * @tparam T Underlying data type of the input buffer
+   * @tparam numDims Number of input dimensions
+   * @param log Logger object
+   */
+  template <typename T, int numDims>
+  void test_type_reinterpret(util::logger& log) {
+    static constexpr size_t inputElemsPerDim = 4;
+    cl::sycl::vector_class<uint8_t> reinterpretInputData(
```
[the remark this answers is not in the document]
Comment (ProGTX) [reply tests/buffer/buffer_api.cpp]: Fixed

**Comment by ProGTX** [comment]
I've also added a small fix for setting `SYCL_LANGUAGE_VERSION` for ComputeCpp: https://github.com/KhronosGroup/SYCL-CTS/pull/47/commits/4c4bdbdc3588b0a6060e9471760fc6fc2a9fe439

**Review by bader — APPROVED** [review]

**Review by keryell — APPROVED** [review]

**Pull request merged**